### PR TITLE
Make KEP-4193 documentation updates

### PR DESCRIPTION
**Pull request by munnerz**
Enhancement: https://github.com/kubernetes/enhancements/issues/4193

### CODE REVIEW

**Comment by chanieljdan** [comment]
Hi @munnerz 👋 just a reminder to take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday March 12th 2024 18:00 PST. 

Thank you!

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 1]
```diff
@@ -60,6 +60,95 @@ for a number of reasons:
...
   usually portable.
 
+## Bound service account tokens
+
+ServiceAccount tokens can be bound to API objects that exist in the kube-apiserver.
+This can be used to tie the validity of a token to the existence of another API object.
+Supported object types are as follows:
+
+* Pod (used for projected volume mounts, see below)
+* Secret (used when generating legacy service account tokens)
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:70]:
```suggestion
* Secret (can be used to allow revoking a token by deleting the Secret)
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 2]
```diff
@@ -60,6 +60,95 @@ for a number of reasons:
...
+* Node (beta in v1.30)
+
+When a token is bound to an object, the objects `metadata.name` and `metadata.uid` are
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:73]:
```suggestion
When a token is bound to an object, the object's `metadata.name` and `metadata.uid` are
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 3]
```diff
@@ -60,6 +60,95 @@ for a number of reasons:
...
+stored as extra 'private claims' in the issued JWT.
+
+When a bound token is presented to the kube-apiserver, the service account authenticator
+will extra and verify these claims.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:77]:
```suggestion
will extract and verify these claims.
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 4]
```diff
@@ -60,6 +60,95 @@ for a number of reasons:
...
+If the referenced object no longer exists (or its `metadata.uid` does not match),
+the request will not be authenticated.
+
+### Additional metadata in Pod bound tokens
+
+{{< feature-state feature_gate_name="ServiceAccountTokenPodNodeInfo" >}}
+
+When a service account token is bound to a Pod object, additional metadata is also
+embedded into the token that indicates the name of the Node that the Pod is scheduled to.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:86]:
```suggestion
embedded into the token that indicates the value of the bound pod's `spec.nodeName` field,
and the uid of that Node, if available.
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 5]
```diff
@@ -60,6 +60,95 @@ for a number of reasons:
...
+
+This node information is **not** verified by the kube-apiserver, and is included to
+avoid integrators needing having to retrieve a full `Pod` object from the apiserver
+when inspecting a JWT.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:90]:
```suggestion
This node information is **not** verified by the kube-apiserver when the token is used for authentication.
It is included so integrators do not have to fetch Pod or Node API objects to check the associated Node name
and uid when inspecting a JWT.
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 6]
```diff
@@ -60,6 +60,95 @@ for a number of reasons:
...
+metadata:
+  creationTimestamp: null
+spec:
+  token: <token>
+status:
+  audiences:
+  - https://kubernetes.default.svc.cluster.local
+  authenticated: true
+  user:
+    extra:
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:132]: wouldn't the node name / uid and jti appear here?

Comment (munnerz) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md:132]: If all of the relevant feature gates are enabled, yes. I guess as these are all enabled by default now though, it does make sense to include them here.

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 7]
```diff
@@ -60,6 +60,96 @@ for a number of reasons:
...
+stored as extra 'private claims' in the issued JWT.
+
+When a bound token is presented to the kube-apiserver, the service account authenticator
+will extract and verify these claims.
+If the referenced object no longer exists (or its `metadata.uid` does not match),
+the request will not be authenticated.
+
+### Additional metadata in Pod bound tokens
+
+{{< feature-state feature_gate_name="ServiceAccountTokenPodNodeInfo" >}}
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:83]: does this automatically pull the beta status, or do we have to indicate that? all the other examples I see have to put the version and state in: https://grep.app/search?q=%7B%7B%3C%20feature-state&filter[repo][0]=kubernetes/website&filter[path][0]=content/en/

**Comment by drewhagen** [comment]
/sig auth
/milestone 1.30

Hello @munnerz  👋, it looks like you've already engaged feedback in review of this PR. Great job! Keep this up. 🚀

@enj @liggitt @mikedanese @liggitt @mikedanese
When one of y'all have a moment, please give a review for technical accuracy. A friendly reminder that this PR needs doc review complete by Doc Freeze on **_March 26th 18:00 PT_** to get this into the release.

@chanieljdan Please review this also. Here is a document on the [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/4193-bound-service-account-token-improvements), including the kep.yaml which has information we'll want to spot-check before release. You can also check a preview of the documentation with Netlify above (feel free to DM me on Slack for instructions with this)

I appreciate everyone's help and so will our users! Thank you greatly.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-jti.md [hunk 8]
```diff
@@ -9,6 +9,9 @@ stages:
   - stage: alpha 
     defaultValue: false
     fromVersion: "1.29"
+  - stage: beta
+    defaultValue: true
+    fromVersion: "v1.30"
```
Comment (drewhagen) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-jti.md:14]:
```suggestion
    fromVersion: "1.30"
```
I think this is why Netfliy deployment preview is failing.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-node-binding-validation.md [hunk 9]
```diff
@@ -9,6 +9,9 @@ stages:
   - stage: alpha 
     defaultValue: false
     fromVersion: "1.29"
+  - stage: beta
+    defaultValue: true
+    fromVersion: "v1.30"
```
Comment (drewhagen) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-node-binding-validation.md:14]:
```suggestion
    fromVersion: "1.30"
```
I think this is why Netfliy deployment preview is failing.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-pod-node-info.md [hunk 10]
```diff
@@ -9,6 +9,9 @@ stages:
   - stage: alpha 
     defaultValue: false
     fromVersion: "1.29"
+  - stage: beta
+    defaultValue: true
+    fromVersion: "v1.30"
```
Comment (drewhagen) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-pod-node-info.md:14]:
```suggestion
    fromVersion: "1.30"
```
I think this is why Netfliy deployment preview is failing.

**Review by drewhagen — COMMENTED** [review]
I just noticed that the Netlify deployment has failed, I think because the correct version should be without the "v"

**Comment by liggitt** [comment]
> @enj @liggitt @mikedanese @liggitt @mikedanese
> When one of y'all have a moment, please give a review for technical accuracy.

I did a pass at https://github.com/kubernetes/website/pull/45292#pullrequestreview-1932599265 and just had a couple outstanding comments

**Comment by chanieljdan** [comment]
Hi @munnerz 👋! I'm reaching out from the Docs team. Just checking in with a reminder as we approach Docs Freeze on Tuesday March 26th 18:00 PDT. This documentation appears to still be under review. To meet the Docs Freeze, this PR must have a technical review as well as lgtm and approve labels applied, without any unaddressed comments or concerns from SIG Docs. 

@kubernetes/sig-docs-en-reviews could we get a review on this PR from a docs perspective?

**Comment by sftim** [comment]
> @kubernetes/sig-docs-en-reviews could we get a review on this PR from a docs perspective?

We can, but: it's still marked as a work in progress. @munnerz: please change the PR title if this is actually ready for review, that will help signal to people that it needs action.

**Comment by munnerz** [comment]
Done - I will address the remaining comments tomorrow morning (UK time) 😊

File: content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-jti.md (hunk 8 above)
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-jti.md:14]: Please consider adding the `toVersion` in the previous stage to align with consistency across other feature gate files. Additionally, the current `fromVersion` value (v1.30) is not in correct format, causing the PR [build to fail](https://app.netlify.com/sites/kubernetes-io-vnext-staging/deploys/65f18b217efc2a0008dfa949). Please consider the following suggestion:


```suggestion
    toVersion: "1.30"
  - stage: beta
    defaultValue: true
    fromVersion: "1.30"
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-node-binding-validation.md (hunk 9 above)
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-node-binding-validation.md:14]: Please consider adding the `toVersion` in the previous stage to align with consistency across other feature gate files. Additionally, the current `fromVersion` value (v1.30) is not in correct format, causing the PR [build to fail](https://app.netlify.com/sites/kubernetes-io-vnext-staging/deploys/65f18b217efc2a0008dfa949). Please consider the following suggestion:


```suggestion
    toVersion: "1.30"
  - stage: beta
    defaultValue: true
    fromVersion: "1.30"
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-pod-node-info.md (hunk 10 above)
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-pod-node-info.md:14]: Please consider adding the `toVersion` in the previous stage to align with consistency across other feature gate files. Additionally, the current `fromVersion` value (v1.30) is not in correct format, causing the PR [build to fail](https://app.netlify.com/sites/kubernetes-io-vnext-staging/deploys/65f18b217efc2a0008dfa949). Please consider the following suggestion:


```suggestion
    toVersion: "1.29"
  - stage: beta
    defaultValue: true
    fromVersion: "1.30"
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 7 above)
Comment (dipesh-rawat) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md:83]: The current proposed change looks good. It leverages the new `feature_gate_name` option which automates the feature status retrieval based on the information in the feature description file i.e. '[service-account-token-pod-node-info.md](https://github.com/kubernetes/website/blob/main/content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-pod-node-info.md?plain=1)'.

Comment (sftim) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md:83]: (see https://kubernetes.io/docs/contribute/new-content/new-features/)

**Comment by munnerz** [comment]
I believe this is all good to go now if you could do another pass 😊

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 11]
```diff
@@ -60,6 +60,102 @@ for a number of reasons:
...
 
+## Bound service account tokens
+
+ServiceAccount tokens can be bound to API objects that exist in the kube-apiserver.
+This can be used to tie the validity of a token to the existence of another API object.
+Supported object types are as follows:
+
+* Pod (used for projected volume mounts, see below)
+* Secret (can be used to allow revoking a token by deleting the Secret)
+* Node (beta in v1.30)
```
Comment (munnerz) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:71]: One finicky part here - authenticating with node bound tokens is beta in 1.30, but actually issuing node bound tokens doesn't go beta until 1.31..

**Comment by chanieljdan** [comment]
Thanks @munnerz!

@dipesh-rawat @liggitt could we get the updates reviewed?

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 11 above)
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:71]:
```suggestion
* Node (in v1.30, creating new node-bound tokens is alpha, using existing node-bound tokens is beta)
```

**Review by liggitt — COMMENTED** [review]
suggested one tweak on the node-bound line, lgtm otherwise

**Comment by chanieljdan** [comment]
Thanks everyone! 

@kubernetes/sig-docs-en-owners does this PR look good for approval?

**Comment by reylejano** [comment]
Thank you @munnerz for addressing comments
/label tide/merge-method-squash
/approve

**Pull request merged**